### PR TITLE
Improve test samples to catch more errors.

### PR DESCRIPTION
**Pull request by ysono**
This PR changes some samples to correctly catch some errors as intended.

I edited `expected_results.json` manually, without running `generate_expected_results.py`. Will `.github/workflows/check_autogen_files.yml` run for this PR?

One thing I left untouched: `tests/chapter_6/valid/ternary_rh_binop.c`: this test passes on solutions that prioritize either the logic or the conditional operator. Maybe using something other than `||` operator on the RHS would be better?

### CODE REVIEW

**Comment by nlsandler** [comment]
This is great, thank you! `.github/workflows/check_autogen_files.yml` will run for this PR, but it should pass - the change you made to `expected_results.json` looks identical to what the script would have changed.

Thanks for flagging the issue in `ternary_rh_binop.c` - I'll open a separate PR to fix that.

**Review by nlsandler — CHANGES_REQUESTED** [review]
Looks good overall - just one question and one spot where we need to update a comment to match your change

**Comment by ysono** [comment]
As with the other PR, please feel free to make amendments. I think you have far better context for how the final change should look like.

File: tests/chapter_11/valid/implicit_casts/common_type.c [hunk 1]
```diff
@@ -28,15 +28,15 @@ int division(void) {
      * the range of int.)
 
      * If instead we truncated l to an int before performing division,
-     * the result would be 1 / 10, or 0.
+     * the result would be -2147483647 / 10, or -214748364.
```
[the remark this answers is not in the document]
Comment (nlsandler) [reply tests/chapter_11/valid/implicit_casts/common_type.c:31]: Ah, you're absolutely right and I made a stupid arithmetic mistake. Thank you!

**Review by nlsandler — APPROVED** [review]

**Pull request merged**